transport_ofi.c: use of undeclared identifier DTYPE_FORTRAN_INTEGER

See https://travis-ci.org/ParRes/Kernels/jobs/127437786 for details.

```c
transport_ofi.c:171:3: error: use of undeclared identifier
      'DTYPE_FORTRAN_INTEGER'
  SHM_INTERNAL_FORTRAN_INTEGER
  ^
./transport_ofi.h:160:38: note: expanded from macro
      'SHM_INTERNAL_FORTRAN_INTEGER'
#define SHM_INTERNAL_FORTRAN_INTEGER DTYPE_FORTRAN_INTEGER
```

`--disable-fortran` seems like the smoking gun here.

--- a/config/shmem_check_type.m4
+++ b/config/shmem_check_type.m4
@@ -55,7 +55,7 @@ AC_DEFUN([SHMEM_FIND_INT_TYPE],[
                 ;;
 	esac
 
-        AC_DEFINE_UNQUOTED(AS_TR_CPP([DTYPE_$1]), [$type], [Portals datatype corresponding to $1])
+        AC_DEFINE_UNQUOTED(AS_TR_CPP([DTYPE_$1]), [$type], [Transport layer datatype corresponding to $1])
 ])
 
 
@@ -104,7 +104,7 @@ AC_DEFUN([SHMEM_FIND_FORTRAN_INT_TYPE],[
 
 	AC_DEFINE_UNQUOTED([DTYPE_FORTRAN_]m4_translit(m4_bpatsubst(m4_bpatsubst([$1], [*], []), [[^a-zA-Z0-9_]], [_]), [a-z], [A-Z]),
                        [$type],
-		       [Portals datatype corresponding to $1])
+		       [Transport layer datatype corresponding to $1])
 	AS_VAR_POPDEF([size_var])
 ])
 
@@ -125,7 +125,7 @@ AC_DEFUN([SHMEM_FIND_FORTRAN_FLOAT_TYPE],[
         esac
 	AC_DEFINE_UNQUOTED([DTYPE_FORTRAN_]m4_translit(m4_bpatsubst(m4_bpatsubst([$1], [*], []), [[^a-zA-Z0-9_]], [_]), [a-z], [A-Z]),
                        [$type],
-		       [Portals datatype corresponding to $1])
+		       [Transport layer datatype corresponding to $1])
 	AS_VAR_POPDEF([size_var])
 ])
 

--- a/configure.ac
+++ b/configure.ac
@@ -301,6 +301,7 @@ else
   FORTRAN_INT8_KIND=0
   FORTRAN_REAL4_KIND=0
   FORTRAN_REAL8_KIND=0
+  AC_DEFINE([DTYPE_FORTRAN_INTEGER],[-1],[Fortran integer C type (disabled)])
 fi
 if test "$ac_cv_sizeof_voidp" = "4" ; then
   FORTRAN_PTR_KIND=$FORTRAN_INT4_KIND

